src/borg/archive.py

```diff
@@ -1053,19 +1053,19 @@ def chunker(it, size):
         pi = ProgressIndicatorPercent(total=count, msg="Verifying data %6.2f%%", step=0.01)
         for chunk_infos in chunker(self.chunks.iteritems(), 100):
             chunk_ids = [chunk_id for chunk_id, _ in chunk_infos]
-            chunk_ids_iter = iter(chunk_ids)
             chunk_data_iter = self.repository.get_many(chunk_ids)
-            while True:
+            while chunk_ids:
                 pi.show()
+                chunk_id = chunk_ids.pop(0)
```

Comment:
What I find telltale in these test results is that it reports _every_ chunk as corrupted. But the test prep only corrupts *one* chunk. Since it says that the chunk IDs don't match it would look like the two iterators are not in sync. My guess is that the chunk data iterator leads by one, thereby is exhausted one too early -> StopIteration. The one chunk that was actually corrupted correctly shows up as an envelope failure.